src/core/connection.c

```diff
@@ -7207,27 +7232,55 @@ QuicConnParamGet(
     }
 
     case QUIC_PARAM_CONN_ORIG_DEST_CID:
+
         if (Connection->OrigDestCID == NULL) {
             Status = QUIC_STATUS_INVALID_STATE;
             break;
         }
+
         if (*BufferLength < Connection->OrigDestCID->Length) {
             Status = QUIC_STATUS_BUFFER_TOO_SMALL;
             *BufferLength = Connection->OrigDestCID->Length;
             break;
         }
+
         if (Buffer == NULL) {
             Status = QUIC_STATUS_INVALID_PARAMETER;
             break;
         }
+
         CxPlatCopyMemory(
             Buffer,
             Connection->OrigDestCID->Data,
             Connection->OrigDestCID->Length);
+
         //
         // Tell app how much buffer we copied.
         //
         *BufferLength = Connection->OrigDestCID->Length;
+
+        Status = QUIC_STATUS_SUCCESS;
+        break;
+
+     case QUIC_PARAM_CONN_SEND_DSCP:
+
+        if (*BufferLength < sizeof(uint8_t)) {
+            Status = QUIC_STATUS_BUFFER_TOO_SMALL;
+            *BufferLength = sizeof(uint8_t);
+            break;
+        }
+
+        if (Buffer == NULL) {
```

Comment:
I see you're just following the pattern in this function, but IMO if a [trusted] BufferLength parameter is > 0, it should imply the buffer pointer is not null.

Reply:
Right, but nothing has validated prior to this function that the buffer is not NULL (discovered via testing). So the validation must happen here.

Reply:
From an SDL [security] perspective it might be worthwhile relocating all these scattered checks to the top of the function: `if (BufferLength > 0 && Buffer == NULL) return ERROR_INVALID_PARAM;` and make it more difficult to have errors of omission in each case statement.

Reply:
I agree, and was expecting to find that already to be the case. However, the API contract allows calling the API with a NULL buffer and the buffer length field set to 0 to obtain the amount of buffer needed to receive the parameter. I don't think that is compatible with an entry-point check of `if (BufferLength > 0 && Buffer == NULL)`

Reply:
Ahhh, this is where so many APIs take subtly different solutions. This is nearly orthogonal to your PR, so I'll move on.